persistence/javadsl/src/main/scala/com/lightbend/lagom/javadsl/persistence/PersistentEntity.scala

```diff
@@ -218,30 +216,75 @@ abstract class PersistentEntity[Command, Event, State] {
       _state = state
     }
 
+    def setEventHandler[A <: Event](eventClass: Class[A], handler: JFunction[A, State]): Unit =
+      setEventHandlerChangingBehavior[A](eventClass, new JFunction[A, Behavior] {
+        override def apply(evt: A): Behavior = behavior.withState(handler.apply(evt))
+      })
+
+    def setEventHandlerChangingBehavior[A <: Event](eventClass: Class[A], handler: JFunction[A, Behavior]): Unit =
+      eventHandlers = eventHandlers.updated(eventClass, handler)
+
+    def removeEventHandler(eventClass: Class[_ <: Event]): Unit =
+      eventHandlers -= eventClass
+
+    def setCommandHandler[R, A <: Command with ReplyType[R]](
+      commandClass: Class[A],
+      handler:      JBiFunction[A, CommandContext[R], Persist[_ <: Event]]
+    ): Unit = {
+      commandHandlers = commandHandlers.updated(
+        commandClass,
+        handler.asInstanceOf[JBiFunction[A, CommandContext[Any], Persist[_ <: Event]]]
+      )
+    }
+
+    def removeCommandHandler(commandClass: Class[_ <: Command]): Unit =
+      commandHandlers -= commandClass
+
+    def setReadOnlyCommandHandler[R, A <: Command with ReplyType[R]](
+      commandClass: Class[A],
+      handler:      JBiConsumer[A, ReadOnlyCommandContext[R]]
+    ): Unit = {
+      setCommandHandler[R, A](commandClass, new JBiFunction[A, CommandContext[R], Persist[_ <: Event]] {
+        override def apply(cmd: A, ctx: CommandContext[R]): Persist[Event] = {
+          handler.accept(cmd, ctx)
+          ctx.done()
+        }
+      });
+    }
+
+    def build(): Behavior =
+      Behavior(_state, eventHandlers, commandHandlers)
+
+  }
+
+  /**
+   * Mutable builder that is used for defining the event and command handlers.
+   * Use [[BehaviorBuilder#build]] to create the immutable [[PersistentEntity.Behavior]].
+   */
+  // In order to provide javadoc preventing instantiation or extension this sealed trait is added
+  // to hold docs and BehaviorBuilderImpl is made private to hold implementation
+  sealed trait BehaviorBuilder {
+    def getState(): State
+    def setState(state: State): Unit
```

Comment:
(Restoring a comment that became "outdated" when I pushed a change to the branch)

No docs for `getState`/`setState`? To be honest, it's not clear to me why you would need these.

Reply:
I think `getState`/`setState` are wrong. We should remove them on a separate PR.

Reply:
Maybe we should remove it on this PR since we are here.

Reply:
Hmm, I see. This was part of previous code and we moved it to the trait. 
I guess we need to leave it as is and eventually get rid of it when we have Akka Persistence Typed.